samcli/lib/utils/tar.py

```diff
@@ -79,9 +79,22 @@ def _validate_destinations_exists(tar_paths: List[Union[str, Path]]) -> bool:
     """
     for file in tar_paths:
         file_path_obj = Path(file)
-        resolved_path = file_path_obj.resolve()
 
-        if file_path_obj.is_symlink() and not resolved_path.exists():
+        try:
+            resolved_path = file_path_obj.resolve()
+        except OSError:
+            # this exception will occur on Windows and will return
+            # a WinError 123
+            LOG.warning("Failed to determine the file on the system")
```

Comment:
Would it be more helpful to Log the path that it failed to resolve, since this single function will validate a list of `tar_paths`.

```suggestion
            LOG.warning("Failed to resolve the path %s on the system", file_path_obj.abspath())
```